src/screens/category/Category.jsx

```diff
@@ -0,0 +1,171 @@
+import axios from 'axios'
+import React, { useEffect, useLayoutEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { Redirect, useHistory } from 'react-router-dom'
+import { categoryDelete, categoryUpdate, createCategory, listCategories } from '../../actions/categoryActions'
+import BackButton from '../../components/backButton/BackButton'
+import Button from '../../components/button/Button'
+import { InputFields } from '../../components/formUI/FormUI'
+import CollectionModalHeader from '../../components/newsCreateModal/CollectionModalHeader'
+import Table from '../../components/table/Table'
+import DashboardLayout from '../../layout/dashboardLayout/DashboardLayout'
+import { configFunc } from '../../utils/apiFunc'
+import './Category.scss'
+import GetRole from '../../utils/getRole.jsx'
+
+const Category = ({ role }) => {
+  const [deleteModal, setDeleteModal] = useState(false)
+  const [editModal, setEditModal] = useState(false)
+  const [addModal, setAddModal] = useState(false)
+  const [deleteId, setDeleteId] = useState(null)
+  const [editId, setEditId] = useState(null)
+  const [data, setData] = useState(null)
+
+  const { loading, categories } = useSelector(state => state.listCategories)
+  const { success } = useSelector(state => state.addCategory)
+  const { success: deleteSuccess } = useSelector(state => state.categoryDelete)
+  const { success: updateSuccess } = useSelector(state => state.categoryUpdate)
+
+  const history = useHistory()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(listCategories())
+  }, [success, deleteSuccess, updateSuccess])
+
+  const config = configFunc()
```

Comment:
this is not required to be used here should use getApi function instead of axios.get